src/backend/marsha/core/views.py

```diff
@@ -397,6 +408,158 @@ class DocumentView(BaseLTIView):
+        """Build app data for the frontend with information retrieved from the LTI launch request.
+
+        Returns
+        -------
+        dictionary
+            Configuration data to bootstrap the frontend:
+            A form is build in react to post data to the LTI Consumer thnough an iframe.
+
+            For instructors only
+            ++++++++++++++++++++
+
+            - lti_select_form_action_url: URL to post data to.
+            - lti_select_form_data: Data to post.
+            - new_document_url: LTI URL to a new document.
+            - new_video_url: LTI URL to a new video.
+            - documents: Documents list with theirs LTI URLs.
+            - videos: Videos list with theirs LTI URLs.
+
+        """
+        documents = DocumentSelectLTISerializer(
+            get_selectable_resources(Document, lti),
+            many=True,
+            context={"request": self.request},
+        ).data
+
+        videos = VideoSelectLTISerializer(
+            get_selectable_resources(Video, lti),
+            many=True,
+            context={"request": self.request},
+        ).data
+
+        new_uuid = str(uuid.uuid4())
+        app_data = _get_base_app_data()
+        app_data.update(
+            {
+                "frontend": "LTI",
+                "lti_select_form_action_url": reverse("select_lti_view"),
+                "lti_select_form_data": self.request.POST,
+                "new_document_url": self.request.build_absolute_uri(
+                    reverse("document_lti_view", args=[new_uuid])
+                ),
+                "new_video_url": self.request.build_absolute_uri(
+                    reverse("video_lti_view", args=[new_uuid])
+                ),
+                "documents": documents,
+                "videos": videos,
+            }
+        )
+        return app_data
+
+    # pylint: disable=unused-argument
+    def post(self, request, *args, **kwargs):
+        """Respond to POST request.
+
+        Populated with context retrieved by get_context_data in the LTI launch request.
+
+        Parameters
+        ----------
+        request : Request
+            passed by Django
+        args : list
+            positional extra arguments
+        kwargs : dictionary
+            keyword extra arguments
+
+        Returns
+        -------
+        HTML
+            generated from applying the data to the template
+
+        """
+        if "content_items" in self.request.POST:
+            lti_select_form_action_url = self.request.POST.get(
+                "content_item_return_url"
+            )
+            # send only needed parameters
+            lti_parameters = {
+                key: value
+                for (key, value) in self.request.POST.items()
+                if key in ("lti_version", "content_items", "data")
+            }
+            lti_parameters["lti_message_type"] = "ContentItemSelection"
+            # generate signature
+            lti = LTI(self.request)
```

Comment:
I think you should instantiate the "lti" object earlier and use it to get the value of ""content_item_return_url", "lti_version", "content_items" and "data" as I understand they are standard lti parameters?

Reply:
The idea behind lti.sign_post_request(url, data) is to be generic (sign any response).
content_items is related to a ContentItemSelectionRequest,  which is specific.